azimuth/modules/base_classes/dask_module.py

```diff
@@ -233,6 +233,9 @@ def wait(self):
         if self.done_event is not None:
             self.done_event.wait()
 
+    def is_pending(self):
+        return self.status() == "pending"
```

Comment:
I think we have to check for the status `"finished"` too, since the task could be `"finished"` but not `done()` because some callbacks have yet to complete.

Also, I think this method should do both `not task.done() and not task.is_pending()`, to avoid repeating that (non so straightforward) logic in two places. For example, it could be:
```suggestion
    def started(self):
        return self.status() in {"pending", "finished"} or self.done()
```
and we call `if not task.started()`, then `task.start...`.

What do you think?

Reply:
I like putting everything in the same method, but I struggle with naming the method `started`, for the use case where it’s in the cache. Especially that the status will be "not started" in that case. 
- Maybe we make explicit, `done_or_pending`? 
- What about we reverse the logic and call the method `should_be_launched`, or something like that?

Reply:
I like `should_be_launched`! 👍

Reply:
Or let's keep `started` instead of `launched`, since the action that follows is `task.start_task...`.